packages/plugin-webpack/plugin.js

```diff
@@ -222,33 +233,40 @@ module.exports = function plugin(config, args) {
         const entrypoints = stats.toJson({ assets: false, hash: true })
           .entrypoints;
 
-        //Now that webpack is done, modify the html file to point to the newly compiled resources
+        //Now that webpack is done, modify the html files to point to the newly compiled resources
         Object.keys(entries).forEach((name) => {
-          const originalScriptEl = entries[name].script;
           if (entrypoints[name] !== undefined && entrypoints[name]) {
             const assetFiles = entrypoints[name].assets || [];
             const jsFiles = assetFiles.filter((d) => d.endsWith(".js"));
             const cssFiles = assetFiles.filter((d) => d.endsWith(".css"));
-            for (const jsFile of jsFiles) {
-              const scriptEl = dom.window.document.createElement("script");
-              scriptEl.src = path.posix.join(baseUrl, jsFile);
-              insertAfter(scriptEl, originalScriptEl);
-            }
-            for (const cssFile of cssFiles) {
-              const linkEl = dom.window.document.createElement("link");
-              linkEl.setAttribute("rel", "stylesheet");
-              linkEl.href = path.posix.join(baseUrl, cssFile);
-              dom.window.document.querySelector("head").append(linkEl);
+
+            for (const occurrence of entries[name].occurrences) {
+              const originalScriptEl = occurrence.script;
+              const dom = occurrence.dom;
+
+              for (const jsFile of jsFiles) {
+                const scriptEl = dom.window.document.createElement("script");
+                scriptEl.src = path.posix.join(baseUrl, jsFile);
+                insertAfter(scriptEl, originalScriptEl);
+              }
+              for (const cssFile of cssFiles) {
+                const linkEl = dom.window.document.createElement("link");
+                linkEl.setAttribute("rel", "stylesheet");
+                linkEl.href = path.posix.join(baseUrl, cssFile);
+                dom.window.document.querySelector("head").append(linkEl);
```

Comment:
Can we move this out of the nested for loop so we don't have to query it every time?

Reply:
done